indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -269,11 +302,90 @@ public void setConf(Configuration config)
     }
   }
 
-  public static class IndexGeneratorReducer extends Reducer<BytesWritable, Writable, BytesWritable, Text>
+  public static class IndexGeneratorCombiner extends Reducer<BytesWritable, BytesWritable, BytesWritable, BytesWritable>
+  {
+    protected HadoopDruidIndexerConfig config;
+    private AggregatorFactory[] combiningAggs;
+
+    @Override
+    protected void setup(Context context)
+        throws IOException, InterruptedException
+    {
+      config = HadoopDruidIndexerConfig.fromConfiguration(context.getConfiguration());
+
+      List<AggregatorFactory> combiningAggsList = Lists.newArrayList();
+      for (AggregatorFactory factory : config.getSchema().getDataSchema().getAggregators()) {
+        combiningAggsList.add(factory.getCombiningFactory());
+      }
+      combiningAggs = combiningAggsList.toArray(new AggregatorFactory[0]);
+    }
+
+    @Override
+    protected void reduce(
+        final BytesWritable key, Iterable<BytesWritable> values, final Context context
+    ) throws IOException, InterruptedException
+    {
+
+      Iterator<BytesWritable> iter = values.iterator();
+      BytesWritable first = iter.next();
+
+      if(iter.hasNext()) {
```

Comment:
I believe this test was originally checking for "is there only one entry for this row, i.e. do I need to combine anything?"  but, if I'm understanding this code, it is actually asking "is there only a single row for this bucket coming from this mapper", which is likely to be false all the time.  If we stay with the current method of doing combining, I don't think this if statement is buying us much.